### PR TITLE
Refactor: Composite component generation (Sourcery) and UserConsentView

### PR DESCRIPTION
**Pull request by billzhou0223**
1. Support declaring child view model inside base view model for composite component (Solve the problem of duplicated protocol conformance)
2. Refactor UserConsent and EULAView
3. Fix scrolling issue in some onboarding screens
4. Introduce AttributedText for taking `NSAttributedString` as input. (Text with AttributedString is only available on iOS 15+)
5. Fix errors in existing examples caused by new Sourcery changes.
6. Update readme.

TODO:
- Fix old UserConsent examples in test app. (Grace will be taking care of it.)

### CODE REVIEW

**Comment by billzhou0223** [comment]
@MarcoEidinger Readme is updated. Please have a review when you have time. Thanks.

**Review by MarcoEidinger — COMMENTED** [review]
Hey Bill, the PR is huge that diff is struggling :) from what I can see (probably I have overlooked some cool stuff) => you did an amazing job. Kudos!

**Major (discussion)**: you changed the code generation so that the properties of a component protocol no longer have the suffix `_`.

I believe this was originally introduced to avoid conflicts when an app developer tries to conform his/her existing model to a component protocol (`invalid redeclaration).

For example, an OData proxy class (the data model) might already define `subtitle` of non-optional type `String` while the component protocol (`SubtitleComponent`) requires the type to be optional (`String?`).

**Minor**: please adjust code example in `sourcery/README.md#sourcery-code-generation` as it is now outdated.

**Comment by MarcoEidinger** [comment]
Also, I don't know if you want to squash or rebase but one conventional commit message should contain footer `BREAKING CHANGE: <short explanation>` so that generated CHANGELOG will highlight that

File: Sources/FioriSwiftUICore/Views/UserConsentView/_UserConsentFormsContainer.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// Generated using Sourcery 1.1.1 — https://github.com/krzysztofzablocki/Sourcery
+// DO NOT EDIT
+import SwiftUI
+
+/// Not used by developers.
+public struct _UserConsentFormsContainer {
```
[the remark this answers is not in the document]
Comment (billzhou0223) [reply Sources/FioriSwiftUICore/Views/UserConsentView/_UserConsentFormsContainer.swift:6]: This type is a requirement for a public extension so it has to be public. I added underscore to the name to indicate that it is not supposed to be used by developer directly. Similar to the case where Apple introduced `_ConditionalContent`.

```
extension UserConsentView where UserConsentForms == _UserConsentFormsContainer {

    public init(userConsentForms: [UserConsentFormModel] = [], didAllow: ((Int) -> Void)? = nil, didDeny: ((Int, Bool) -> Void)? = nil, didCancel: ((Int) -> Void)? = nil, didFinish: (([Int]) -> Void)? = nil) {
        self._userConsentForms = _UserConsentFormsContainer(userConsentForms: userConsentForms)
    }
}
```

**Comment by billzhou0223** [comment]
> Hey Bill, the PR is huge that diff is struggling :) from what I can see (probably I have overlooked some cool stuff) => you did an amazing job. Kudos!
> 
> **Major (discussion)**: you changed the code generation so that the properties of a component protocol no longer have the suffix `_`.
> 
> I believe this was originally introduced to avoid conflicts when an app developer tries to conform his/her existing model to a component protocol (`invalid redeclaration).
> 
> For example, an OData proxy class (the data model) might already define `subtitle` of non-optional type `String` while the component protocol (`SubtitleComponent`) requires the type to be optional (`String?`).
> 
> **Minor**: please adjust code example in `sourcery/README.md#sourcery-code-generation` as it is now outdated.

This is a very interesting point.

> Hey Bill, the PR is huge that diff is struggling :) from what I can see (probably I have overlooked some cool stuff) => you did an amazing job. Kudos!
> 
> **Major (discussion)**: you changed the code generation so that the properties of a component protocol no longer have the suffix `_`.
> 
> I believe this was originally introduced to avoid conflicts when an app developer tries to conform his/her existing model to a component protocol (`invalid redeclaration).
> 
> For example, an OData proxy class (the data model) might already define `subtitle` of non-optional type `String` while the component protocol (`SubtitleComponent`) requires the type to be optional (`String?`).
> 
> **Minor**: please adjust code example in `sourcery/README.md#sourcery-code-generation` as it is now outdated.

This is a very interesting argument. 

I think key debate here is if it's fair for us to ask app developers to conform to the requirements we have in our protocols and have them resolve the conflicts by modifying their implementation on their side. Or we should anticipate what developers already have and try to design our APIs in a way to avoid those conflicts.

In my opinon, it happens more often that as SDK consumers they need to observe the rules set by SDK. You rarely see in other frameworks (UIKIt, etc) that they use the name with `_` suffix in public interfaces. And we don't take this approach in other SAP SDKs as well. For example `FUIUser` represents the users added in multi-user onboarding. It is declared as follows:

```
public protocol FUIUser {
    /**
     The image for the user.
    */
    var image: UIImage? { get }

    /**
     The user name.
    */
    var name: String { get }

    /**
     This property could be email, employee ID, or other user information.
     */
    var infoString: String? { get }

    /**
     An ID string to identify the user.
     */
    var id: String { get }
}
```

This is definitely debatable so let me know what you think. I can revert it to the old name convension if it makes more sense to you :)

BTW the readme in Sourcery is updated per request, please have another review.

**Review by MarcoEidinger — APPROVED** [review]
Ok, let's change the component protocol names to not use the `_` suffix. 

My reasoning (in addition to your argument): Name clashes are now more likely (than before with the `suffix`) in theory but practically I believe such clashes will be the exception. The app developer can wrap his/her model and adopt the wrapper to conform to component protocols. This might be any way the case if the models are structs and app developer wants to build observable view models.

The message of merging commit message contain footer BREAKING CHANGE: <short explanation> so that generated CHANGELOG will highlight that. But we can also create CHANGELOG later manually. Next release version should be 2.0 due to this major change

**Pull request merged**